### DISCUSSION TITLE
Rocky / OSGEarth

### DISCUSSION THREAD

**Discussion by gd-6** [pelicanmapping/rocky#57] [Q&A / Support] [answered]
Hi,

I'm very interested in your solution, but I have a few questions I'd like to clarify.

Do you have an estimated release date for the first stable version?

How will Rocky differ from OSG Earth? Will it offer similar functionalities but be based on VSG, or should we expect significant changes, with features being added, removed, or modified?

If I want to start a project involving Earth maps soon, would you recommend using OSG Earth or Rocky at this stage?

Also, do you have any plans to support CityGML or IFC file formats in the future?

Thank you for all your hard work!

**Comment by gwaldron** [comment, marked as answer]
Why? As you know osgEarth is based upon the OpenSceneGraph SDK. New feature development has ceased for OSG and it's in maintenance mode. OSG itself it based on OpenGL which is in a similar situation; we don't really expect OpenGL to support new hardware features as they come out since all future focus will be on Vulkan.

Rocky is a new C++ terrain rendering API using a similar data model to osgEarth, but able to move forward along with Vulkan.

Yes, it makes sense to consider it if you're starting a new project. But it is not a "replacement" for osgEarth _per se_. Backwards compatibility and feature compatibility with osgEarth are not our goals.

As of this writing Rocky is supported by a small amount of R&D funding as a potential long-term replacement for osgEarth as OSG and OpenGL slowly fade into the background. There is no timeline for a stable release, but as always this will be driven by the community and the funding opportunities :) It will be helpful to get feedback on what a "Minimum Viable 1.0" will need to support!

Hope this makes sense. Happy to answer any and all questions.
